### ISSUE TITLE
Weird behaviour when expanding nodes

### ISSUE THREAD

**Issue by d2fong** [iVis-at-Bilkent/cytoscape.js-expand-collapse#43]
## Example
1.  I collapse complex nodes
2.  I run a layout
3.  Expand a complex node
4.  The position the children nodes had before the layout will be vastly different from where their parent node is now.  

I am about to click the blue node to expand it using the extension api:

![screen shot 2017-07-06 at 3 16 27 pm](https://user-images.githubusercontent.com/2328291/27928730-56e67ee0-625e-11e7-9b7f-30d7b947a9b3.png)

Here is where it ends up after I click to expand it:
![screen shot 2017-07-06 at 3 16 33 pm](https://user-images.githubusercontent.com/2328291/27928791-897c19f0-625e-11e7-87aa-a6e1513967db.png)

For example it would be nice to specify position for the child nodes like this, so when it expands, I can pass in it's parents position so it doesn't move to unexpected positions:

```js
api.expand(node, { childPos: {x: 100, y: 100}});
```

I currently bind an event to track the parent's position, and set the children's position after the expand, but it still looks bad @gbader @ugurdogrusoz

**Comment by ugurdogrusoz** [comment]
@d2fong This shouldn't happen even if you turn off "fisheye" and "layout" (see the behavior in Newt for instance). How do you initialize the extension? Have you looked at the settings described here: https://docs.google.com/document/d/16EX9IBmJyYmPqxbCon-r4qE-5nDWl2SB6g9-nGApkO0/edit?usp=sharing

**Comment by d2fong** [comment]
@ugurdogrusoz @gbader

Yes I am following that document and using all of the settings you recommend for collapsing complexes by default.

**Comment by ugurdogrusoz** [comment]
Since the nodes have not moved at all after expand, layoutBy must be null (default); can you please try setting it to cose-bilkent with randomize "false" and animate set to "end"? If that doesn't work, I suggest we hold a hangout sometime next week to go over your code and track the problem down.

**Comment by d2fong** [comment]
Are you sure it is not a problem with the extension?

**Comment by ugurdogrusoz** [comment]
I am not claiming it's not a bug on our end; I am just trying to understand why we're not able to reproduce it on our end, with the settings suggested. Perhaps, you're setting animate to "during" instead of "end" since there is a known issue for this: https://github.com/cytoscape/cytoscape.js-cose-bilkent/issues/47? This wouldn't explain why nodes wouldn't change locations at all (except the expanded node) upon expand since there should be an incremental layout running during expand. Can we please have a hangout where you show us the related code and we try to track it down on Monday? Are you available your afternoon?

**Comment by d2fong** [comment]
Okay good to know you couldn't reproduce it.  I'll investigate more.

Thanks.

**Comment by d2fong** [comment]
Okay, I found the issues with my code:

1.  I wasn't using the latest version of the extension.
2.  I was using the expandcollapse.afterexpand event which I assumed was fired after the whole operation was complete, but it is actually fired before the expanded node children's positions are updated, which lead to the behaviour that I described in the github extension.

Thanks again for the support @ugurdogrusoz
